### PR TITLE
FIX: match all possible space separator to add quotes

### LINKED ISSUE #87: Issue with separator space other than space
We had a dataset with the character `\u3000` which is one of the separator space. The regular expression in the encoder only matches `\\` while it should use `\s` to match all possible separators and add quotes around the string to be compliant with the ARFF format.

I will open a PR to solve this issue.

### PR DESCRIPTION
**Pull request by glemaitre**
closes #87 

Matches all possible separator spaces to add quotes.

NB: seen in a dataset with Japanese character with an ideographic space (`\u3000`)

### CODE REVIEW

**Comment by glemaitre** [comment]
@mfeurer Can you have a look at this.

File: CHANGES.rst [hunk 1]
```diff
@@ -1,6 +1,10 @@
 What's New on LIAC-ARFF
 =======================
 
+LIAC-ARFF 2.3.2
+* fix: match all possible separator spaces to add quotes to be compliant with
```
Comment (jnothman) [inline CHANGES.rst]: When reading? Writing?

File: arff.py [hunk 2]
```diff
@@ -164,7 +164,7 @@
 _RE_RELATION     = re.compile(r'^([^\{\}%,\s]*|\".*\"|\'.*\')$', re.UNICODE)
 _RE_ATTRIBUTE    = re.compile(r'^(\".*\"|\'.*\'|[^\{\}%,\s]*)\s+(.+)$', re.UNICODE)
 _RE_TYPE_NOMINAL = re.compile(r'^\{\s*((\".*\"|\'.*\'|\S*)\s*,\s*)*(\".*\"|\'.*\'|\S*)\s*\}$', re.UNICODE)
-_RE_QUOTE_CHARS = re.compile(r'["\'\\ \t%,]')
+_RE_QUOTE_CHARS = re.compile(r'["\'\s%,]', re.UNICODE)
```
Comment (jnothman) [inline arff.py]: Do we have a test to ensure this will not capture newlines inappropriately?

Comment (mfeurer) [reply arff.py]: First, if I understand this correctly, `\s` should capture newlines, right? Second, if `\n` and `\r`are defined as whitespace it should be fine to match them here as this only encodes single values.

**Comment by mfeurer** [comment]
Looks good to me. I'd merge once the comment from @jnothman is resolved and the description is clearer. Also, could you please add yourself to the list of contributors?

File: arff.py (hunk 2 above)
Comment (glemaitre) [reply arff.py]: I think that this is fine on the encoding side. The decoding is however the place which is going wrong (#89).

File: CHANGES.rst [hunk 3]
```diff
@@ -1,6 +1,10 @@
...
+  ARFF format when writing.
```
Comment (jnothman) [inline CHANGES.rst]: You've said "writing" here but "The decoding is however the place which is going wrong" in your comment.

Comment (glemaitre) [reply CHANGES.rst]: Oh my comment was really misleading. What I meant about this comment was the following.
The encoding of the `\r\n` and `\n` into ARFF file is well handled (with quote around the string). The issue with `\r\n` and `\n` is happening during the ARFF decoding which as we discuss in #89

Comment (mfeurer) [reply CHANGES.rst]: Could you please fix the formatting of your what's new message?

**Comment by glemaitre** [comment]
Whoops! Done.

**Comment by jnothman** [comment]
LGTM

**Comment by jnothman** [comment]
But you might want to just add this to the features I've got proposed for 2.4 (#85, #91)

**Pull request merged**